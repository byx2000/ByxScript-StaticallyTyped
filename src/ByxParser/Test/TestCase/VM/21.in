int g1;
int g2;
int g3;
int g4;
int g5;
int g6;

function int fib(int n)
{
	if (n != 1)
	{
		if (n != 2)
		{
			return fib(n - 1) + fib(n - 2);
		}
		else
		{
			return 1;
		}
	}
	else
	{
		return 1;
	}
}

function int f(int n)
{
	if (n == 1)
	{
		if (n == 2)
		{
			return 1;
		}
		else
		{
			return 2;
		}
	}
	else
	{
		if (n == 3)
		{
			return 3;
		}
		else
		{
			return 4;
		}
	}
}

function void g(int n)
{
	if (n < 255)
	{
		g6 = 568723;
		return;
	}
	g6 = 272829;
}

function void main()
{
	g1 = fib(25);
	g2 = f(1);
	g3 = f(2);
	g4 = f(3);
	g5 = f(4);
	g(123);
}