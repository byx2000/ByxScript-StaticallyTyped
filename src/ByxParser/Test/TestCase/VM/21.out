i 75025
i 2
i 4
i 3
i 4
i 568723